admin/app/components/solidus_admin/users/addresses/component.html.erb

```diff
@@ -0,0 +1,56 @@
+<%= page do %>
+  <%= page_header do %>
+    <%= page_header_back(solidus_admin.users_path) %>
+    <%= page_header_title(t(".title", email: @user.email)) %>
+
+    <%= page_header_actions do %>
+      <%= render component("ui/button").new(tag: :a, text: t(".create_order_for_user"), href: spree.new_admin_order_path(user_id: @user.id)) %>
+    <% end %>
+  <% end %>
+
+  <%= page_header do %>
+    <% tabs.each do |tab| %>
+      <%= render(component("ui/button").new(tag: :a, scheme: :ghost, text: tab[:text], 'aria-current': tab[:current], href: tab[:href])) %>
+    <% end %>
+  <% end %>
+
+  <%= page_with_sidebar do %>
+    <%= page_with_sidebar_main do %>
+      <%= render component('ui/panel').new(title: t(".billing_address")) do %>
+        <%= form_for @user, url: solidus_admin.update_addresses_user_path(@user), method: :put, html: { id: "#{form_id}_billing", autocomplete: "off", class: "bill_address" } do |f| %>
+
+          <%= render component('ui/forms/address').new(address: (@user.bill_address || Spree::Address.build_default), name: "user[bill_address_attributes]") %>
+          <div class="py-1.5 text-center">
+            <%= render component("ui/button").new(tag: :button, text: t(".update"), form: "#{form_id}_billing") %>
+            <%= render component("ui/button").new(tag: :a, text: t(".cancel"), href: solidus_admin.addresses_user_path(@user), scheme: :secondary) %>
+          </div>
+        <% end %>
+      <% end %>
+
+      <%= render component('ui/panel').new(title: t(".shipping_address")) do %>
+        <%= form_for @user, url: solidus_admin.update_addresses_user_path(@user), method: :put, html: { id: "#{form_id}_shipping", autocomplete: "off", class: "ship_address"  } do |f| %>
+
+          <%= render component('ui/forms/address').new(address: (@user.ship_address || Spree::Address.build_default), name: "user[ship_address_attributes]") %>
```

Comment:
Is this `"ui/forms/address"` component not dealing with validation errors? I saw in your video that the field is not marked as erroneous and I would expect the validation error message to appear below the field. It doesn't seem to be the case, though.

Another thing I found was that we need to deal with [`Spree::Config.address_requires_state`](https://github.com/solidusio/solidus/blob/3a92a45e94940641e9b4392c9a20d1241f651ce9/core/lib/spree/app_configuration.rb#L38). The [current admin hides and shows the state select](https://github.com/solidusio/solidus/blob/main/backend/app/assets/javascripts/spree/backend/addresses.js) depending on that setting.

I would like to have this behavior in the new admin as well.

Reply:
It does deal with validation and mark the form accordingly as long as the validation errors are generated from the back-end. As you can see in the video (and by my controller) I had to work around it a bit as there were no errors when trying to update the user even when the address was not provided valid values, which I assumed came from the lack of `accepts_nested_attributes_for`. So I am not sure if the inclusion of the `UserAddressBook` is working as expected?

Reply:
Ok @tvdeyen I did check the inclusion, and it looks like you are correct that devise includes `UserMethods` which in turn includes `Spree::UserAddressBook`, so we _should_ have access to those validations. Is there something I am doing wrong here that is resulting in me not getting those error messages on the user? Why is it "successfully updating" the user even when the address is being rejected for missing the name and it's falling back to the values it had before the update?

Currently the only reason we have errors on the `@address` object is because I am choosing to explicitly build and validate that object. It's not a natural or required part of this controller action, nor was it there before (hence the total lack of validation on the legacy backend address admin).

I have attached a quick exploratory video of the back-end, but if you have any additional info you could share that would be really helpful. Taking another look at the additional `def set_address_from_params` method I added would be great too as I'd like some feedback on the back-end stuff that is happening.

https://github.com/user-attachments/assets/a7d0180c-da62-4b89-bc16-174f9f922256

Reply:
Ok disregard above and see newly attached videos!